packages/ckeditor5-dev-release-tools/lib/utils/executeinparallel.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* eslint-env node */
+
+'use strict';
+
+const crypto = require( 'crypto' );
+const path = require( 'path' );
+const fs = require( 'fs' );
+const { Worker } = require( 'worker_threads' );
+const glob = require( 'glob' );
+const { tools } = require( '@ckeditor/ckeditor5-dev-utils' );
+
+const WORKER_SCRIPT = path.join( __dirname, 'parallelworker.js' );
+
+/**
+ * This util allows executing a specified task in parallel using Workers. It can be helpful when executing a not resource-consuming
+ * task in all packages specified in a path.
+ *
+ * If the callback loads dependencies, they must be specified directly in the function due to the worker's limitations.
+ * Functions cannot be passed to workers. Hence, we store the callback as a Node.js file loaded by workers.
+ *
+ * @see https://nodejs.org/api/worker_threads.html
+ * @param {Object} options
+ * @param {String} options.packagesDirectory
+ * @param {String} options.processDescription
+ * @param {Function} options.callback
+ * @param {AbortSignal} options.signal
+ * @param {String} [options.cwd=process.cwd()]
+ * @param {Number} [options.concurrency=require( 'os' ).cpus().length / 2]
+ * @returns {Promise}
```

Comment:
```suggestion
 * @param {Object} options
 * @param {String} options.packagesDirectory Relative path to a location of packages to execute a task.
 * @param {String} options.processDescription Description of the task displayed on a screen.
 * @param {Function} options.taskToExecute A callback that is executed on all found packages. It receives an absolute path to a package as an argument. It can be synchronous or may return a promise.
 * @param {AbortSignal} options.signal Signal to abort the asynchronous process.
 * @param {String} [options.cwd=process.cwd()] Current working directory from which all paths will be resolved.
 * @param {Number} [options.concurrency=require( 'os' ).cpus().length / 2] Number of CPUs that will execute the task.
 * @returns {Promise}
```